a.go

```diff
@@ -315,19 +355,27 @@ func (a *A) runCleanups(finished *bool, panicVal *interface{}, panicStack *[]byt
 		}
 	}()
 
-	for {
-		var cleanup func()
-		a.mu.Lock()
-		if len(a.cleanups) > 0 {
-			last := len(a.cleanups) - 1
-			cleanup = a.cleanups[last]
-			a.cleanups = a.cleanups[:last]
-		}
-		a.mu.Unlock()
-		if cleanup == nil {
-			cleanupFinished = true
-			return
-		}
-		cleanup()
+	a.callCleanups()
+	cleanupFinished = true
+}
+
+func (a *A) callLastCleanup() bool {
+	var cleanup func()
+	a.mu.Lock()
+	if len(a.cleanups) > 0 {
+		last := len(a.cleanups) - 1
+		cleanup = a.cleanups[last]
+		a.cleanups = a.cleanups[:last]
+	}
+	a.mu.Unlock()
+	if cleanup == nil {
```

Comment:
I probably discovered an already existing bug (please not to fix it in this PR).
If an action calls `a.Cleanup(nil)` then will would prevent previously registered to not being called.
Am I correct? If so then we can fix it in a separate PR.

Reply:
I think this should be something more or less like

```go
	for {
		var cleanup func()
		a.mu.Lock()
		if len(a.cleanups) == 0 {
			a.mu.Unlock()
			cleanupFinished = true
			return
		}
		last := len(a.cleanups) - 1
		cleanup = a.cleanups[last]
		a.cleanups = a.cleanups[:last]
		a.mu.Unlock()
		cleanup()
	}
```

Reply:
> I didn't try to fix this problem, and if I did, it wasn't on purpose :) 

No. You have not fixed or introduced a problem. It is an exploratory discovery during review :sweat_smile: 
I will make an issue out of my comment.

Reply:
https://github.com/goyek/goyek/issues/454